addons/mrp/models/stock_rule.py

```diff
@@ -73,7 +73,24 @@ def _run_manufacture(self, procurements):
                     domain += (('procurement_group_id', '=', group.id),)
                 mo = self.env['mrp.production'].sudo().search(domain, limit=1)
             if not mo:
-                new_productions_values_by_company[procurement.company_id.id]['values'].append(rule._prepare_mo_vals(*procurement, bom))
+                if procurement.values.get('batch_size', False):
+                    total_qty = procurement.product_qty
+                    while total_qty > 0:
+                        qty = min(total_qty, procurement.values['batch_size'])
```

Comment:
maybe just ensure batch_size is always positive

Reply:
Alright :+1: I added a constrain in enterprise.

Reply:
I think it's better to add it here since it could be used later for other usage than MPS